src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,74 @@
+import { concatOneIssueList } from "../../issueHandling/issueMapping";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+
+export enum Mode {
+	propositionalLogic = "diproche",
+	firstOrderPredicateLogic = "diproche_fo",
+	test = "teste",
+}
+
+/**
+ * Since the user input most likely lacks the correct predicate,
+ * it needs to be added. For easier maintainability in case
+ * of an expanding predicate list, the corresponding predicate is
+ * added with respect to the current @param mode. This can, for
+ * example correspond to the concept of "Spielwiese". The user should
+ * be able to just stay inside a mode and not have to add this mode
+ * himself. Hence, the mode needs to be read out from the browser.
+ */
+export function addPredicate(userInput: string, mode: Mode) {
+	return mode.concat("(").concat(userInput).concat(").");
+}
+
+// Assuming "wrongWord" is not an allowed word, then a
+// vocaberror is an error like "the word "wrongWord"
+// in the text "This text contains a wrongWord".
+export function getVocabErrors(userInput: string): void {
+	concatOneIssueList(getAllIssues(userInput));
```

Comment:
You should use the `addIssue`-method here. Since that method takes the issue-code only though, you will need to change `getAllIssues` to not return issues, but add the issue-codes via `addIssue` directly.

Reply:
I don't understand why it would be disadventageous to immediately append a whole Issue[] to another one. If I understand the reason behind your suggestion I would agree on what I have to change, though.